File: C++/557. Reverse Words in a String III.cpp
// Time:  O(n)
// Space: O(n)
class Solution {
public:
    string reverseWords(string s) {
        int last = 0;
        for(int i = 0;i<(int)s.length();i++)
        {
            if(i==s.length()-1 || s[i+1]==' ')
            {
                for(int j=last,k=i;j<k;j++,k--)
                    swap(s[k],s[j]);
                last = i+2;
                //cout<<s<<endl;
            }
        }
        return s;
    }
};

File: Python/557. Reverse Words in a String III.py
# Time:  O(n)
# Space: O(n)
class Solution:
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        ans = s.split(' ')
        for i in range(len(ans)):
            ans[i] = ans[i][::-1]
        ans = ' '.join(ans)
        return ans
             